tests/playwright-test/playwright.expect.true.spec.ts

```diff
@@ -352,6 +352,35 @@ test('should support toBeFocused', async ({ runInlineTest }) => {
   expect(result.exitCode).toBe(0);
 });
 
+test.only('should support toBeFocused with shadow elements', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'a.test.ts': `
+      const { test } = pwt;
+
+      test('focused', async ({ page }) => {
+        await page.setContent(\`
+          <div id="app">
+          </div>
+          <script>
+            const root = document.querySelector('div');
+            const shadowRoot = root.attachShadow({ mode: 'open' });
+            const input = document.createElement('input');
+            input.id = "my-input"
+            shadowRoot.appendChild(input);
+          </script>
+        \`);
+
+        await page.locator("input").focus();
+        expect(await page.evaluate(() => document.activeElement.shadowRoot.activeElement.id)).toBe("my-input");
+        await expect(page.locator("#app")).toBeFocused();
+        await expect(page.locator("input")).toBeFocused();
```

Comment:
NB: These are two different elements that are focused. I think this aligns with our definition of this assertion, and intuitively makes sense.